### PR TITLE
Team47 quiz creation

### PR DESCRIPTION
**Pull request by dlahtine**
Created an endpoint for professors to create quizzes.

to test: make a post request to `http://localhost:8080/createQuiz` with the following JSON:
`
{"title":"How to test an endpoint",
 "course_id":1,
 "instructions":"These are the instructions",
 "shuffle":true,
 "time_limit":9000,
 "date_open":"1999-01-01",
 "date_close":"2020-01-01",
 "quiz_type":"quiz",
 "attempts":90,
 "quiz_group":"Computer Science",
 "total_points":100,
 "questions":[
   {
     "quesType": "MC",
     "points": 50,
     "content": "This quiz sucks.",
     "choices": [
       {
         "correct": false,
         "content": "True"
       },
       {
         "correct": true,
         "content": "False"
       }
     ]
   },
   {
     "quesType": "MA",
     "points": 50,
     "content": "Select C and D",
     "choices": [
       {
         "correct": false,
         "content": "A"
       },
       {
         "correct": false,
         "content": "B"
       },
       {
         "correct": true,
         "content": "C"
       },
       {
         "correct": true,
         "content": "D"
       }
     ]
   }
 ]
}`

Then check if it made it into the db.
Also try using invalid datatypes for the different fields, or leave out necessary fields to test that you recieve an error response.

### CODE REVIEW

File: Team_47/QuizCreator/src/main/java/com/asu/ser516/team47/servlet/QuizCreationServlet.java [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+                return;
+            }
+        } catch (ClassCastException ex) {
+            res.sendError(400, "Invalid format for dates.");
+            return;
+        } catch (NullPointerException npe) {
+            //Date not included: open immediately and use no close date.
+        }
+        if (date_open == null) date_open = new Date();
+        Quiz qz = new Quiz(GARBAGE_VALUE, title, course_id, instructions, shuffle, time_limit, date_open, date_close,
```
Comment (PaHorton) [inline Team_47/QuizCreator/src/main/java/com/asu/ser516/team47/servlet/QuizCreationServlet.java]: title and course_id need to be switched. As it stands this does not compile.

File: Team_47/QuizCreator/src/main/java/com/asu/ser516/team47/servlet/SubmissionServlet.java [hunk 2]
```diff
@@ -82,30 +74,25 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
...
             }
-            enrollId = ((Long)requestForm.get("enrolled_id")).intValue();
+            enrollId = ((Number)requestForm.get("enrolled_id")).intValue();
             enrollment = new EnrolledDAOImpl().getEnrolled(enrollId);
             if (enrollment == null) {
                 httpCode = 500;
             }
             JSONArray jsonChoices = (JSONArray) requestForm.get("choices");
-            choiceIds = ServletValidation.buildAndValidateChoiceList(jsonChoices, quizId);
+            choiceIds = ServletValidation.buildAndValidateSubmittedChoiceList(jsonChoices, quizId);
```
Comment (PaHorton) [inline Team_47/QuizCreator/src/main/java/com/asu/ser516/team47/servlet/SubmissionServlet.java]: This method does not exist in ServletValidation

**Review by PaHorton — CHANGES_REQUESTED** [review]
I found two syntax errors

File: Team_47/QuizCreator/src/main/java/com/asu/ser516/team47/servlet/QuizCreationServlet.java (hunk 1 above)
Comment (dlahtine) [reply Team_47/QuizCreator/src/main/java/com/asu/ser516/team47/servlet/QuizCreationServlet.java]: That is embarrasing; I thought I compiled this before making a PR.

**Review by HalcyonAura — CHANGES_REQUESTED** [review]
Works well, just remove the extra print statements and next time (or even comment here) add a way to explicitly test the db instead of relying on the tester to write a query after the fact/read the db file. LMK when print stuff is removed and I'll approve.

**Review by HalcyonAura — APPROVED** [review]

**Review by PaHorton — APPROVED** [review]
After the changes, this looks good to me!

**Comment by dlahtine** [comment]
Merging in from master has people on edge, me included. moving changes to a new branch.

**Pull request closed without merging**